includes/abstracts/abstract.llms.database.query.php

```diff
@@ -27,158 +27,69 @@ abstract class LLMS_Database_Query {
-
-	/**
-	 * Array of results retrieved by the query.
-	 *
-	 * @var array
-	 */
-	public $results = array();
-
-	/**
-	 * The raw SQL query.
-	 *
-	 * @var string
-	 */
-	protected $sql = '';
-
-	/**
-	 * Constructor.
-	 *
-	 * @since 3.8.0
-	 *
-	 * @param array $args Optional. Query arguments. Default empty array.
-	 *                    When not provided the default arguments will be used.
-	 * @return void
+	 * @return array
 	 */
-	public function __construct( $args = array() ) {
-
-		$this->arguments_original = $args;
-		$this->arguments_default  = $this->get_default_args();
+	protected function default_arguments() {
 
-		$this->setup_args();
-
-		$this->query();
+		return wp_parse_args(
+			array(
+				'per_page' => 25,
+				'sort'     => array(
+					'id' => 'ASC',
+				),
+			),
+			parent::default_arguments()
+		);
 
 	}
 
 	/**
 	 * Escape and add quotes to a string, useful for array mapping when building queries.
 	 *
 	 * @since 3.8.0
+	 * @since [version] Use {@see llms_esc_and_quote_str()}.
 	 *
 	 * @param mixed $input Input data.
 	 * @return string
 	 */
 	public function escape_and_quote_string( $input ) {
-		return "'" . esc_sql( $input ) . "'";
-	}
-
-	/**
-	 * Retrieve a query variable with an optional fallback / default.
-	 *
-	 * @since 3.8.0
-	 *
-	 * @param string $key     Variable key.
-	 * @param mixed  $default Default value.
-	 * @return mixed
-	 */
-	public function get( $key, $default = '' ) {
-
-		if ( isset( $this->query_vars[ $key ] ) ) {
-			return $this->query_vars[ $key ];
-		}
-
-		return $default;
+		return llms_esc_and_quote_str( $input );
 	}
 
 	/**
 	 * Retrieve default arguments for the query.
 	 *
 	 * @since 3.8.0
 	 * @since 4.5.1 Added new default arg `no_found_rows` set to false.
+	 * @since [version] Call parent method.
+	 *
+	 * @todo This should be removed in favor of the parent method only when the
+	 *       `llms_db_query_get_default_args` hook is removed.
 	 *
 	 * @return array
 	 */
 	protected function get_default_args() {
 
-		$args = array(
-			'page'             => 1,
-			'per_page'         => 25,
-			'search'           => '',
-			'sort'             => array(
-				'id' => 'ASC',
-			),
-			'suppress_filters' => false,
-			'no_found_rows'    => false,
-		);
-
 		if ( $this->get( 'suppress_filters' ) ) {
-			return $args;
+			return $this->default_arguments();
 		}
 
+		// Get the from the parent with the new replacement filter.
+		$args = parent::get_default_args();
+
 		/**
 		 * Filters the query default args.
 		 *
 		 * @since 3.8.0
+		 * @deprecated [version] Filter `llms_db_query_get_default_args` is deprecated in favor of `llms_{$this->id}_query_get_default_args`.
 		 *
 		 * @param array $args Array of default arguments to set up the query with.
 		 */
-		return apply_filters( 'llms_db_query_get_default_args', $args );
+		return apply_filters_deprecated( 'llms_db_query_get_default_args', array( $args ), '[version]', 'llms_{$this->id}_query_get_default_args' );
```

Comment:
```suggestion
		return apply_filters_deprecated( 'llms_db_query_get_default_args', array( $args ), '[version]', "llms_{$this->id}_query_get_default_args" );
```
did you mean this, right?

Reply:
I didn't, because I was thinking since it was a dynamic filter/string it should output as such but that doesn't make sense actually, since the dev using the filter would need to lookup the id instead of us just giving it to them.